<?xml version="1.0" ?>

<container xmlns="http://symfony.com/schema/dic/services"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">

    <parameters>
        <parameter key="secotrust.sabredav.controller.class">Secotrust\Bundle\SabreDavBundle\Controller\SabreDavController</parameter>
        <parameter key="secotrust.sabredav.exceptionHandler.class">Symfony\Component\ErrorHandler\ErrorRenderer\HtmlErrorRenderer</parameter>
        <parameter key="secotrust.sabredav.server.inner.class">Sabre\DAV\Server</parameter>
        <parameter key="secotrust.sabredav.server.class">Secotrust\Bundle\SabreDavBundle\SabreDav\Server</parameter>
        <parameter key="secotrust.sabredav.use_symfony_exception_handler">1</parameter>
        <parameter key="secotrust.sabredav.server.nodes" type="collection">
            <parameter id="secotrust.sabredav_principal_collection" />
            <parameter id="secotrust.sabredav_carddav_collection" />
            <parameter id="secotrust.sabredav_caldav_collection" />
        </parameter>
    </parameters>

    <services>
        <service id="secotrust.sabredav.server" class="%secotrust.sabredav.server.class%">
            <argument type="service" id="secotrust.sabredav.server.inner" />
            <argument type="service" id="secotrust.sabredav.exceptionHandler" />
            <argument type="service" id="router" />
            <argument>%secotrust.sabredav.use_symfony_exception_handler%</argument>
        </service>
        <service id="secotrust.sabredav.server.inner" class="%secotrust.sabredav.server.inner.class%">
            <argument id="secotrust.sabredav.server.nodes" />
        </service>
        <service id="secotrust.sabredav.exceptionHandler" class="%secotrust.sabredav.exceptionHandler.class%">
            <argument key="$debug">true</argument>
        </service>

    </services>
</container>
